datastore/elastic.go

```diff
@@ -0,0 +1,181 @@
+package datastore
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/bar41234/bar_book_service/models"
+	"github.com/olivere/elastic/v7"
+	"strconv"
+	"strings"
+)
+
+const (
+	ElasticUrl = "http://es-search-7.fiverrdev.com:9200"
+	IndexName  = "bar_books_store_2"
+	QuerySize  = 100
+)
+
+type ElasticDb struct {
+	client *elastic.Client
+	isInit bool
+}
+
+func (e *ElasticDb) Init() error {
+	if !e.isInit {
+		client, err := elastic.NewClient(elastic.SetURL(ElasticUrl))
+		if err != nil {
+			return err
+		}
+		e.client = client
+		ctx := context.Background()
+		_, _, err = e.client.Ping(ElasticUrl).Do(ctx)
+		if err != nil {
+			return err
+		}
+		e.isInit = true
+	}
+	return nil
+}
+
+func (e *ElasticDb) Get(id string) (*models.Book, error) {
+	initErr := e.Init()
+	if initErr != nil {
+		return nil, initErr
+	}
+	ctx := context.Background()
+	get, getErr := e.client.Get().Index(IndexName).Id(id).Do(ctx)
+	if getErr != nil {
+		return nil, getErr
+	}
```

Comment:
```suggestion
	bookResponse, err := e.client.Get().Index(IndexName).Id(id).Do(ctx)
	if err != nil {
		return nil, err
	}
```
or just `response` \ `res`

Reply:
Can I use err variable more than once at the same scope? (first usage with :=, others with = )

Reply:
yep!